Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -508,6 +565,9 @@ namespace Multiplayer
             AZ_PROFILE_SCOPE(MULTIPLAYER, "MultiplayerSystemComponent: OnTick - SendReliablePackets");
             m_networkInterface->GetConnectionSet().VisitConnections(visitor);
         }
+
+        const auto duration = AZStd::chrono::duration_cast<AZStd::chrono::microseconds>(AZStd::chrono::steady_clock::now() - startMultiplayerTickTime);
+        stats.RecordFrameTime(AZ::TimeUs{ duration.count() });
```

Comment:
I think there should be type safe timer wrappers that will avoid you needing to cast between time types here.. AZ::ITime::GetRealElapsedTimeUs() and friends, although I admit the documentation is quite terrible here.. I have no idea what the difference is between all of RealSimulationTickTime, RealElapsedTime, ElapsedTime, ElapsedTimeDebug, SimulationTime, etc.. as an example; it looks like folks have been adding a kitchen sink worth of concepts without describing what they all are or why we need them.

Reply:
We don't want any kind of simulation time here since the actual elapsed time is what needs to be measured.
It looks like that is [GetRealElapsedTimeUs](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Time/TimeSystem.cpp#L103-L110).

Reply:
I wish we have a profile class that would measure this time for me.

Reply:
Looking at:
``` cpp
    TimeUs TimeSystem::GetRealElapsedTimeUs() const
    {
        const TimeUs currentTime = static_cast<TimeUs>(AZStd::GetTimeNowMicroSecond());
        m_realAccumulatedTimeUs += currentTime - m_realLastInvokedTimeUs;
        m_realLastInvokedTimeUs = currentTime;

        return m_realAccumulatedTimeUs;
    }
```

This is not safe to call to get a valid results since if something calls in between start and end calls.

Reply:
The [duration begin and duration end](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.nso4gcezn7n1) events are paired up and would measure the time for you.

Reply:
You mean, that the processing of stream output would have to compute that? It seems easier to capture as one event for the first pass.

Reply:
I mean there is two events for measuring a duration of time(the duration begin and end event) that should paired together.
There is also the complete event that combines the duration begin and end event into a single call, but the user is required to supply the duration in that case(https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.lpfof2aylapb)